[vcpkg_acquire_msys,vcpkg-ci-msys2] Ensure msys2 package list freshness

dg0yt opened the pull request:
I was waiting for an embarassing incident like the recent removal of the outdated libtool package used by vcpkg from the rolling-release servers of msys2...
Edit: Next incident: msys2-runtime, #30542, https://github.com/microsoft/vcpkg/issues/30530, https://github.com/microsoft/vcpkg/pull/30546.
Edit: Next incident: python-numpy, #30582. (However, the incident affects tensorflow-cc which is not tested in CI. The issue might not be fixed by this PR.)
Edit: Next incident: grep, #31446, https://github.com/ObjectVision/GeoDMS/issues/225.
Edit: msys2-runtime is still an issue in VS, #31565, #31577, two months after the fix in vcpkg!

This PR is an attempt to help prevent such incidents.
- It adds a test port which gives a report for msys2 package versions and dependencies.
  - Reporting covers `vcpkg_acquire_msys`, `vcpkg_find_acquire_program(PKGCONFIG)` and `vcpkg_find_fortran`.
  - The test port warns or even fails when msys2 packages become too old.
    The check isn't entirely accurate because it would require to know when a package was updated. But this wouldn't matter if updates are done regularly. msys2 packages are removed 1.5 after an update ATM.
  - The test port can update the package URLs and SHA512s if installed with the `update-all` feature.
    (Note: This even works on linux or osx if installed with `--allow-unsupported`.)
    User may decide to line-wise pick only some of the changes using `git gui` or similar.
  - By updating all packages at once, vcpkg would avoid combinations of packages with very different age, matching proper MSYS2 installations more closely.
  - The test port would't be triggered by (i.e. block) most regular PRs. It would always run in full CI. This is intented.
  - Improving the test port doesn't cause expensive rebuilds. So it can be done incrementally after merging this PR.
- It significantly rewrites `vcpkg_acquire_msys`.
  - Packages are not simply installed in a given order, but based on declared dependencies.
    This allows to list the packages in alphabetical order, avoiding multiple definitions with inconsistent dependencies.
  - Downloaded files are now prefixed `msys2`. It is not msys.

dg0yt committed:
[vcpkg-ci-msys2] New test port for msys2 freshness

dg0yt commented:
Current state:
~~~
-- +++ Analyzing package lists
-- unzip
-- - vcpkg deps: libbz2
-- + msys2 deps: bash;libbz2
-- libbz2
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-3-x86_64.pkg.tar.zst (403 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-4-x86_64.pkg.tar.zst (207 days ago)
-- patch
-- - vcpkg: https://repo.msys2.org/msys/x86_64/patch-2.7.6-1-x86_64.pkg.tar.xz (1805 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/patch-2.7.6-2-x86_64.pkg.tar.zst (66 days ago)
-- gzip
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gzip-1.11-1-x86_64.pkg.tar.zst (495 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gzip-1.12-2-x86_64.pkg.tar.zst (207 days ago)
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: bash
-- texinfo
-- - vcpkg: https://repo.msys2.org/msys/x86_64/texinfo-6.8-3-x86_64.pkg.tar.zst (376 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/texinfo-7.0.1-1-x86_64.pkg.tar.zst (47 days ago)
-- bash
-- - vcpkg: https://repo.msys2.org/msys/x86_64/bash-5.1.008-1-x86_64.pkg.tar.zst (610 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/bash-5.2.009-1-x86_64.pkg.tar.zst (72 days ago)
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: 
-- autoconf
-- - vcpkg: https://repo.msys2.org/msys/x86_64/autoconf-2.71-3-any.pkg.tar.zst (375 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/autoconf2.71-2.71-1-any.pkg.tar.zst (368 days ago)
-- - vcpkg deps: m4;perl
-- + msys2 deps: awk;bash;diffutils;m4;perl;sed
-- ! unknown deps: awk
-- autoconf-archive
-- - vcpkg: https://repo.msys2.org/msys/x86_64/autoconf-archive-2019.01.06-1-any.pkg.tar.xz (1466 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/autoconf-archive-2022.09.03-1-any.pkg.tar.zst (48 days ago)
-- - vcpkg deps: m4;perl
-- + msys2 deps: 
-- diffutils
-- - vcpkg: https://repo.msys2.org/msys/x86_64/diffutils-3.8-2-x86_64.pkg.tar.zst (368 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/diffutils-3.8-4-x86_64.pkg.tar.zst (67 days ago)
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: bash;libiconv;libintl
-- binutils
-- - vcpkg: https://repo.msys2.org/msys/x86_64/binutils-2.37-5-x86_64.pkg.tar.zst (396 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/binutils-2.39-2-x86_64.pkg.tar.zst (93 days ago)
-- - vcpkg deps: libiconv;libintl
-- + msys2 deps: libiconv;libintl;zlib
-- libtool
-- - vcpkg deps: coreutils;file;findutils;grep;sed
-- + msys2 deps: bash;tar
-- file
-- - vcpkg: https://repo.msys2.org/msys/x86_64/file-5.41-2-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/file-5.44-2-x86_64.pkg.tar.zst (11 days ago)
-- - vcpkg deps: gcc-libs;libbz2;zlib
-- + msys2 deps: gcc-libs;libbz2;liblzma;zlib
-- zlib
-- - vcpkg: https://repo.msys2.org/msys/x86_64/zlib-1.2.11-1-x86_64.pkg.tar.xz (2186 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/zlib-1.2.13-1-x86_64.pkg.tar.zst (82 days ago)
-- bzip2
-- - vcpkg: https://repo.msys2.org/msys/x86_64/bzip2-1.0.8-3-x86_64.pkg.tar.zst (403 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/bzip2-1.0.8-4-x86_64.pkg.tar.zst (207 days ago)
-- - vcpkg deps: gcc-libs
-- + msys2 deps: libbz2
-- libbz2
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-3-x86_64.pkg.tar.zst (403 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-4-x86_64.pkg.tar.zst (207 days ago)
-- coreutils
-- - vcpkg: https://repo.msys2.org/msys/x86_64/coreutils-8.32-2-x86_64.pkg.tar.zst (454 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/coreutils-8.32-5-x86_64.pkg.tar.zst (65 days ago)
-- grep
-- - vcpkg: https://repo.msys2.org/msys/x86_64/grep-3.0-2-x86_64.pkg.tar.xz (1506 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/grep-1~3.0-6-x86_64.pkg.tar.zst (65 days ago)
-- - vcpkg deps: libiconv;libintl;libpcre
-- + msys2 deps: bash;libiconv;libintl;libpcre
-- sed
-- - vcpkg: https://repo.msys2.org/msys/x86_64/sed-4.8-2-x86_64.pkg.tar.zst (386 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/sed-4.9-1-x86_64.pkg.tar.zst (61 days ago)
-- - vcpkg deps: libintl
-- + msys2 deps: bash;libintl
-- libpcre
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libpcre-8.45-1-x86_64.pkg.tar.zst (580 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libpcre-8.45-3-x86_64.pkg.tar.zst (88 days ago)
-- m4
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: bash;gcc-libs;libiconv
-- automake-wrapper
-- - vcpkg deps: gawk
-- + msys2 deps: automake1.11;automake1.12;automake1.13;automake1.14;automake1.15;automake1.16;bash;gawk
-- ! unknown deps: automake1.11;automake1.12;automake1.13;automake1.14;automake1.15
-- gawk
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gawk-5.1.0-2-x86_64.pkg.tar.zst (403 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gawk-5.2.1-2-x86_64.pkg.tar.zst (35 days ago)
-- - vcpkg deps: libintl;libreadline;mpfr
-- + msys2 deps: bash;libintl;libreadline;mpfr
-- mpfr
-- - vcpkg: https://repo.msys2.org/msys/x86_64/mpfr-4.1.0-1-x86_64.pkg.tar.zst (916 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/mpfr-4.2.0-2-x86_64.pkg.tar.zst (2 days ago)
-- - vcpkg deps: gcc-libs;gmp
-- + msys2 deps: gmp
-- gmp
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gmp-6.2.1-1-x86_64.pkg.tar.zst (760 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gmp-6.2.1-2-x86_64.pkg.tar.zst (276 days ago)
-- xz
-- - vcpkg: https://repo.msys2.org/msys/x86_64/xz-5.2.5-1-x86_64.pkg.tar.xz (1037 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/xz-5.4.0-1-x86_64.pkg.tar.zst (31 days ago)
-- - vcpkg deps: gettext;libiconv;liblzma
-- + msys2 deps: libiconv;libintl;liblzma
-- liblzma
-- - vcpkg: https://repo.msys2.org/msys/x86_64/liblzma-5.2.5-1-x86_64.pkg.tar.xz (1037 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/liblzma-5.4.0-1-x86_64.pkg.tar.zst (31 days ago)
-- - vcpkg deps: 
-- + msys2 deps: bash;gettext;libiconv
-- libreadline
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libreadline-8.1.001-1-x86_64.pkg.tar.zst (543 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libreadline-8.2.001-3-x86_64.pkg.tar.zst (2 days ago)
-- ncurses
-- - vcpkg: https://repo.msys2.org/msys/x86_64/ncurses-6.2-2-x86_64.pkg.tar.zst (403 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/ncurses-6.3-3-x86_64.pkg.tar.zst (265 days ago)
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: gcc-libs
-- automake1.16
-- - vcpkg: https://repo.msys2.org/msys/x86_64/automake1.16-1.16.3-3-any.pkg.tar.zst (379 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/automake1.16-1.16.5-1-any.pkg.tar.zst (276 days ago)
-- - vcpkg deps: perl
-- + msys2 deps: bash;perl
-- perl
-- - vcpkg: https://repo.msys2.org/msys/x86_64/perl-5.32.1-2-x86_64.pkg.tar.zst (383 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/perl-5.36.0-1-x86_64.pkg.tar.zst (85 days ago)
-- - vcpkg deps: libcrypt
-- + msys2 deps: bash;coreutils;db;gdbm;libcrypt
-- ! unknown deps: db;gdbm
-- libcrypt
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libcrypt-2.1-3-x86_64.pkg.tar.zst (647 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libcrypt-2.1-4-x86_64.pkg.tar.zst (89 days ago)
-- pkg-config
-- - vcpkg: https://repo.msys2.org/msys/x86_64/pkg-config-0.29.2-4-x86_64.pkg.tar.zst (814 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/pkgconf-1.8.0-2-x86_64.pkg.tar.zst (66 days ago)
-- - vcpkg deps: libiconv
-- + msys2 deps: 
-- make
-- - vcpkg: https://repo.msys2.org/msys/x86_64/make-4.3-3-x86_64.pkg.tar.zst (374 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/make-4.4-1-x86_64.pkg.tar.zst (48 days ago)
-- - vcpkg deps: libintl;msys2-runtime
-- + msys2 deps: bash;libintl
-- gettext-devel
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gettext-devel-0.21-1-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gettext-devel-0.21-2-x86_64.pkg.tar.zst (250 days ago)
-- - vcpkg deps: gettext
-- + msys2 deps: gettext;libiconv-devel
-- ! unknown deps: libiconv-devel
-- gettext
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gettext-0.21-1-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gettext-0.21-2-x86_64.pkg.tar.zst (250 days ago)
-- - vcpkg deps: libasprintf;libgettextpo;libintl;tar
-- + msys2 deps: libasprintf;libgettextpo;libintl
-- tar
-- - vcpkg: https://repo.msys2.org/msys/x86_64/tar-1.34-2-x86_64.pkg.tar.zst (386 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/tar-1.34-3-x86_64.pkg.tar.zst (207 days ago)
-- - vcpkg deps: libiconv;libintl
-- + msys2 deps: bash;libiconv;libintl
-- libgettextpo
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libgettextpo-0.21-1-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libgettextpo-0.21-2-x86_64.pkg.tar.zst (250 days ago)
-- libasprintf
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libasprintf-0.21-1-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libasprintf-0.21-2-x86_64.pkg.tar.zst (250 days ago)
-- findutils
-- - vcpkg: https://repo.msys2.org/msys/x86_64/findutils-4.8.0-1-x86_64.pkg.tar.zst (657 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/findutils-4.9.0-3-x86_64.pkg.tar.zst (250 days ago)
-- libintl
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libintl-0.21-1-x86_64.pkg.tar.zst (382 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libintl-0.21-2-x86_64.pkg.tar.zst (250 days ago)
-- - vcpkg deps: libiconv
-- + msys2 deps: gcc-libs;libiconv
-- libiconv
-- - vcpkg: https://repo.msys2.org/msys/x86_64/libiconv-1.16-2-x86_64.pkg.tar.zst (863 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/libiconv-1.17-1-x86_64.pkg.tar.zst (227 days ago)
-- - vcpkg deps: gcc-libs
-- + msys2 deps: gcc-libs;libintl
-- gcc-libs
-- - vcpkg: https://repo.msys2.org/msys/x86_64/gcc-libs-11.2.0-3-x86_64.pkg.tar.zst (402 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/gcc-libs-11.3.0-2-x86_64.pkg.tar.zst (198 days ago)
-- - vcpkg deps: msys2-runtime
-- + msys2 deps: 
-- msys2-runtime
-- - vcpkg: https://repo.msys2.org/msys/x86_64/msys2-runtime-3.2.0-8-x86_64.pkg.tar.zst (605 days ago)
-- + msys2: https://repo.msys2.org/msys/x86_64/msys2-runtime-3.4.3-6-x86_64.pkg.tar.zst (2 days ago)
-- which
-- - vcpkg deps: 
-- + msys2 deps: bash
-- mingw-w64-x86_64-python-numpy
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-numpy-1.20.3-1-any.pkg.tar.zst (616 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-numpy-1.23.5-1-any.pkg.tar.zst (60 days ago)
-- mingw-w64-x86_64-openblas
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openblas-0.3.19-1-any.pkg.tar.zst (392 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openblas-0.3.21-7-any.pkg.tar.zst (38 days ago)
-- - vcpkg deps: mingw-w64-x86_64-gcc-libgfortran;mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-libwinpthread
-- + msys2 deps: mingw-w64-x86_64-gcc-libgfortran;mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-libwinpthread;mingw-w64-x86_64-omp
-- ! unknown deps: mingw-w64-x86_64-omp
-- mingw-w64-x86_64-gcc-libgfortran
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libgfortran-11.2.0-8-any.pkg.tar.zst (362 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libgfortran-12.2.0-9-any.pkg.tar.zst (3 days ago)
-- mingw-w64-x86_64-python
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-3.8.9-2-any.pkg.tar.zst (646 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-python-3.10.9-2-any.pkg.tar.zst (5 days ago)
-- mingw-w64-x86_64-bzip2
-- mingw-w64-x86_64-mpdecimal
-- mingw-w64-x86_64-ncurses
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ncurses-6.3-3-any.pkg.tar.zst (404 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ncurses-6.3-6-any.pkg.tar.zst (71 days ago)
-- mingw-w64-x86_64-libsystre
-- mingw-w64-x86_64-libtre
-- mingw-w64-x86_64-openssl
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.m-1-any.pkg.tar.zst (397 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-3.0.7-3-any.pkg.tar.zst (0 days ago)
-- - vcpkg deps: mingw-w64-x86_64-ca-certificates;mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-zlib
-- + msys2 deps: 
-- mingw-w64-x86_64-ca-certificates
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20210119-1-any.pkg.tar.zst (468 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-ca-certificates-20211016-4-any.pkg.tar.zst (2 days ago)
-- mingw-w64-x86_64-p11-kit
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-p11-kit-0.24.1-1-any.pkg.tar.zst (363 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-p11-kit-0.24.1-3-any.pkg.tar.zst (162 days ago)
-- mingw-w64-x86_64-libtasn1
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libtasn1-4.18.0-1-any.pkg.tar.zst (427 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libtasn1-4.19.0-1-any.pkg.tar.zst (148 days ago)
-- mingw-w64-x86_64-sqlite3
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-sqlite3-3.37.2-1-any.pkg.tar.zst (375 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-sqlite3-3.40.1-1-any.pkg.tar.zst (18 days ago)
-- - vcpkg deps: mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-readline;mingw-w64-x86_64-tcl
-- + msys2 deps: mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-readline;mingw-w64-x86_64-tcl;mingw-w64-x86_64-zlib
-- mingw-w64-x86_64-readline
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-readline-8.1.001-1-any.pkg.tar.zst (517 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-readline-8.2.001-6-any.pkg.tar.zst (9 days ago)
-- mingw-w64-x86_64-termcap
-- mingw-w64-x86_64-tk
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tk-8.6.11.1-2-any.pkg.tar.zst (647 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tk-8.6.12-1-any.pkg.tar.zst (117 days ago)
-- mingw-w64-x86_64-tcl
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tcl-8.6.11-5-any.pkg.tar.zst (522 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-tcl-8.6.12-1-any.pkg.tar.zst (117 days ago)
-- mingw-w64-x86_64-xz
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-xz-5.2.5-2-any.pkg.tar.zst (758 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-xz-5.2.9-1-any.pkg.tar.zst (49 days ago)
-- - vcpkg deps: mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-gettext
-- + msys2 deps: mingw-w64-x86_64-gettext
-- mingw-w64-x86_64-gettext
-- mingw-w64-x86_64-gcc-libs
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libs-11.2.0-8-any.pkg.tar.zst (362 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc-libs-12.2.0-9-any.pkg.tar.zst (3 days ago)
-- - vcpkg deps: mingw-w64-x86_64-gmp;mingw-w64-x86_64-libwinpthread;mingw-w64-x86_64-mpc;mingw-w64-x86_64-mpfr
-- + msys2 deps: mingw-w64-x86_64-libwinpthread
-- mingw-w64-x86_64-mpc
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpc-1.2.1-1-any.pkg.tar.zst (752 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpc-1.3.1-1-any.pkg.tar.zst (24 days ago)
-- - vcpkg deps: mingw-w64-x86_64-mpfr
-- + msys2 deps: mingw-w64-x86_64-gmp;mingw-w64-x86_64-mpfr
-- mingw-w64-x86_64-mpfr
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpfr-4.1.0-3-any.pkg.tar.zst (832 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-mpfr-4.2.0-1-any.pkg.tar.zst (13 days ago)
-- - vcpkg deps: mingw-w64-x86_64-gmp
-- + msys2 deps: mingw-w64-x86_64-gcc-libs;mingw-w64-x86_64-gmp
-- mingw-w64-x86_64-gmp
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gmp-6.2.1-3-any.pkg.tar.zst (379 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gmp-6.2.1-4-any.pkg.tar.zst (24 days ago)
-- mingw-w64-x86_64-expat
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-expat-2.4.4-1-any.pkg.tar.zst (355 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-expat-2.5.0-1-any.pkg.tar.zst (77 days ago)
-- mingw-w64-x86_64-libffi
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libffi-3.3-4-any.pkg.tar.zst (648 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libffi-3.4.4-1-any.pkg.tar.zst (83 days ago)
-- mingw-w64-x86_64-libiconv
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libiconv-1.16-2-any.pkg.tar.zst (763 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libiconv-1.17-1-any.pkg.tar.zst (224 days ago)
-- mingw-w64-x86_64-zlib
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-zlib-1.2.11-9-any.pkg.tar.zst (744 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-zlib-1.2.13-2-any.pkg.tar.zst (62 days ago)
-- mingw-w64-x86_64-libwinpthread
-- - vcpkg: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst (379 days ago)
-- + msys2: https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libwinpthread-git-10.0.0.r202.g4359b3570-1-any.pkg.tar.zst (11 days ago)
CMake Warning at scripts/test_ports/vcpkg-ci-msys2/portfile.cmake:209 (message):
  The following msys2 packages have changed dependencies: unzip gzip bash
  autoconf autoconf-archive diffutils binutils libtool file bzip2 grep sed m4
  automake-wrapper gawk mpfr xz liblzma ncurses automake1.16 perl pkg-config
  make gettext-devel gettext tar libintl libiconv gcc-libs which
  mingw-w64-x86_64-openblas mingw-w64-x86_64-openssl mingw-w64-x86_64-sqlite3
  mingw-w64-x86_64-xz mingw-w64-x86_64-gcc-libs mingw-w64-x86_64-mpc
  mingw-w64-x86_64-mpfr
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)


CMake Warning at scripts/test_ports/vcpkg-ci-msys2/portfile.cmake:215 (message):
  The following msys2 packages would be needed to update all dependencies:
  automake1.11 automake1.12 automake1.13 automake1.14 automake1.15 awk db
  gdbm libiconv-devel mingw-w64-x86_64-omp
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)


CMake Error at scripts/test_ports/vcpkg-ci-msys2/portfile.cmake:223 (message):
  The following msys2 packages were build more than 6 months ago and should
  be updated: libbz2 patch gzip texinfo bash autoconf autoconf-archive
  diffutils binutils file zlib bzip2 libbz2 coreutils grep sed libpcre gawk
  mpfr gmp xz liblzma libreadline ncurses automake1.16 perl libcrypt
  pkg-config make gettext-devel gettext tar libgettextpo libasprintf
  findutils libintl libiconv gcc-libs msys2-runtime
  mingw-w64-x86_64-python-numpy mingw-w64-x86_64-openblas
  mingw-w64-x86_64-gcc-libgfortran mingw-w64-x86_64-python
  mingw-w64-x86_64-ncurses mingw-w64-x86_64-openssl
  mingw-w64-x86_64-ca-certificates mingw-w64-x86_64-p11-kit
  mingw-w64-x86_64-libtasn1 mingw-w64-x86_64-sqlite3
  mingw-w64-x86_64-readline mingw-w64-x86_64-tk mingw-w64-x86_64-tcl
  mingw-w64-x86_64-xz mingw-w64-x86_64-gcc-libs mingw-w64-x86_64-mpc
  mingw-w64-x86_64-mpfr mingw-w64-x86_64-gmp mingw-w64-x86_64-expat
  mingw-w64-x86_64-libffi mingw-w64-x86_64-libiconv mingw-w64-x86_64-zlib
  mingw-w64-x86_64-libwinpthread
Call Stack (most recent call first):
  scripts/ports.cmake:147 (include)
~~~

dg0yt commented:
So libtool in vcpkg was updated after removal from msys2 servers... And now we wait for the next package to break? zlib seems a good candidate. The package downloaded by vcpkg is from 2186 days ago - app. 6 years old.

I can try to incrementally improve the situation based on the test ports output.
From my first experiments, this includes refactoring of `vcpkg_acquire_msys`, to enable better match with msys2 dependencies.
- Upstream has circular dependencies (e.g. libintl <-> libiconv). vcpkg could extract the packages in any order, but currently it cannot handle these cases.
- In theory, `vcpkg_acquire_msys` could read dependencies from the package in the same way as the test port does it now. This would simplify updates. However:
  - This also implies handling of "provides" properties (e.g. bash provides sh). These have to be hardcoded if we don't want to search the whole database for providers of missing names.
  - Upstream's dependencies also include package which are not strictly required for the vcpkg use case, e.g. `info`.

The test port currently doesn't cover "direct packages" such as pkg-config (which is x86!) from `vcpkg_find_acquire_program`. However, these packages have the same potential to disrupt vcpkg usage when they are removed from msys2 server.

vcpkg has to be some months ahead of package removal from msys2 servers in order to mitigate effects on user which  aren't on vcpkg `HEAD`.

Neumann-A commented:
> Upstream has circular dependencies (e.g. libintl <-> libiconv). vcpkg could extract the packages in any order, but currently it cannot handle these cases.

It is not really a circular dependency. It is libiconv -> libintl (to get gettext tools) -> libiconv with nls -> libintl with nls. So having just libintl depend on libiconv is ok (since we don't build msys stuff.)

Also why do we have ncurses and python from msys ?

dg0yt commented:
> > Upstream has circular dependencies (e.g. libintl <-> libiconv). vcpkg could extract the packages in any order, but currently it cannot handle these cases.
> 
> It is not really a circular dependency. It is libiconv -> libintl (to get gettext tools) -> libiconv with nls -> libintl with nls. So having just libintl depend on libiconv is ok (since we don't build msys stuff.)

Well, package libintl is really just libintl, depending on libiconv. Tools are in package gettext.
However, package libiconv also has iconv.exe which depends on libintl.

> Also why do we have ncurses and python from msys ?

ncurses is a dependency of readline which is a dependeny of gawk which is... (autotools).
Don't know about mingw-w64-x86_64-python. But once the name is in the scripts, it cannot be removed.

dg0yt committed:
Add features for script updates

dg0yt committed:
Update primary URL and mirrors

dg0yt committed:
Follow repo renames

dg0yt commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -136,361 +135,364 @@ function(vcpkg_acquire_msys out_msys_root)
         DEPS libbz2
     )
     z_vcpkg_acquire_msys_declare_package(
-        URL "https://repo.msys2.org/msys/x86_64/libbz2-1.0.8-3-x86_64.pkg.tar.zst"
-        SHA512 955420cabd45a02f431f5b685d8dc8acbd07a8dcdda5fdf8b9de37d3ab02d427bdb0a6d8b67c448e307f21094e405e916fd37a8e9805abd03610f45c02d64b9e
```
libbz2 was listed twice.

dg0yt commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -1,13 +1,12 @@
     "https://mirror.yandex.ru/mirrors/msys2/"
     "https://mirrors.tuna.tsinghua.edu.cn/msys2/"
     "https://mirrors.ustc.edu.cn/msys2/"
-    "https://mirror.bit.edu.cn/msys2/"
     "https://mirror.selfnet.de/msys2/"
-    "https://mirrors.sjtug.sjtu.edu.cn/msys2/"
```
Removed tier 2 mirrors.
The main switch is now the main URL, mirror.msys2.org.

dg0yt commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -136,361 +135,364 @@ function(vcpkg_acquire_msys out_msys_root)
     z_vcpkg_acquire_msys_declare_package(
-        URL "https://repo.msys2.org/msys/x86_64/autoconf-2.71-3-any.pkg.tar.zst"
-        SHA512 f639deac9b2191c2096584cf374103bbb1e9d2476dd0ebec94b1e80da59be25b88e362ac5280487a89f0bb0ed57f99b66e314f36b7de9cda03c0974806a3a4e2
+        NAME autoconf
+        URL "https://mirror.msys2.org/msys/x86_64/autoconf-wrapper-15-1-any.pkg.tar.zst"
+        SHA512 7c0f0c619100d05c82409567399253efddee9b39fff7dd772e503770afd4accbae2ce96307a20b076faf2e308413cd25e7405e70969608223bc86c7016ec38b8
+        DEPS autoconf2.71
```
New autoconf packaging.

dg0yt commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -136,361 +135,364 @@ function(vcpkg_acquire_msys out_msys_root)
     )
     z_vcpkg_acquire_msys_declare_package(
-        URL "https://repo.msys2.org/msys/x86_64/pkg-config-0.29.2-4-x86_64.pkg.tar.zst"
-        SHA512 9f72c81d8095ca1c341998bc80788f7ce125770ec4252f1eb6445b9cba74db5614caf9a6cc7c0fcc2ac18d4a0f972c49b9f245c3c9c8e588126be6c72a8c1818
+        NAME pkg-config
+        URL "https://mirror.msys2.org/msys/x86_64/pkgconf-1.8.0-2-x86_64.pkg.tar.zst"
+        SHA512 4fcc1671969098b9b7c79192f90d3c0396ad65cb6efc44fdd7e6e7a37452f0bf4a38fb2ef63e73b0f9883a87c9f248c38b7f66c7d762bfbbd0da1d22aec5b52d
```
pkg-config.exe is now provided from package pkgconf.

dg0yt commented:
So this is how libtool looks for the x64-windows import library of ws2_32 and fails:
~~~
 - /D/installed/x64-windows/lib
 - /C/PROGRA~1/MICROS~1/2022/ENTERP~1/VC/Tools/MSVC/1434~1.319/atlmfc/lib/x64
 - /C/PROGRA~1/MICROS~1/2022/ENTERP~1/VC/Tools/MSVC/1434~1.319/lib/x64
 - /C/PROGRA~2/WI3CF2~1/NETFXSDK/4.8/Lib/um/x64
 - /C/PROGRA~2/WI3CF2~1/10/Lib/100226~1.0/ucrt/x64
 - /C/PROGRA~2/WI3CF2~1/10/Lib/100226~1.0/um/x64
!!! '/C/PROGRA~2/WI3CF2~1/10/Lib/100226~1.0/um/x64/WS2_32.Lib'
 - /c/a/1/s/scripts/buildsystems/make_wrapper
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/VC/Tools/MSVC/14.34.31933/bin/HostX64/x64
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/VC/VCPackages
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/TestWindow
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/TeamFoundation/Team
 - Explorer
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/MSBuild/Current/bin/Roslyn
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Team
 - Tools/Performance
 - Tools/x64
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Team
 - Tools/Performance
 - Tools
 - /c/Program
 - Files
 - (x86)/Microsoft
 - SDKs/Windows/v10.0A/bin/NETFX
 - 4.8
 - Tools/x64
 - /c/Program
 - Files
 - (x86)/HTML
 - Help
 - Workshop
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/Extensions/Microsoft/CodeCoverage.Console
 - /c/Program
 - Files
 - (x86)/Windows
 - Kits/10/bin/10.0.22621.0/x64
 - /c/Program
 - Files
 - (x86)/Windows
 - Kits/10/bin/x64
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/MSBuild/Current/Bin/amd64
 - /c/Windows/Microsoft.NET/Framework64/v4.0.30319
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/Tools
 - /c/Program
 - Files/PowerShell/7
 - /c/a/1/s/scripts/buildsystems/make_wrapper
 - /usr/share/automake-1.16
 - /usr/bin
 - /c/Windows/system32
!!! '/c/Windows/system32/ws2_32.dll'
 - /c/Windows
 - /c/Windows/system32/Wbem
 - /c/Windows/system32/WindowsPowerShell/v1.0
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/VC/Tools/Llvm/x64/bin
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/VC/Linux/bin/ConnectionManagerExe
 - /d/downloads/tools/win_bison/2.5.25
 - /c/Program
 - Files/Microsoft
 - Visual
 - Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja
 - /d/downloads/tools/msys2/9a1ec3f33446b195/mingw32/bin

*** Warning: linker path does not have real file for library -lws2_32.
*** I have the capability to make that library automatically link in when
*** you link to this library.  But I can only do this if you have a
*** shared version of the library, which you do not appear to have
*** because I did check the linker path looking for a file starting
*** with ws2_32 and none of the candidates passed a file format test
*** using a file magic. Last file checked: /c/Windows/system32/ws2_32.dll
~~~

- It finds `/C/PROGRA~2/WI3CF2~1/10/Lib/100226~1.0/um/x64/WS2_32.Lib` but doesn't accept it. 
  (Not okay! This is the desired file.) 
  (However we would need a dependency on port `winsock2` because this is that port's reason to exist.)
- It finds `/c/Windows/system32/ws2_32.dll` but doesn't accept it.
  (Okay, but shouldn't reach this point.)
- While libtool properly looks at the library path (even converted to DOS conventions to resolve spaces), the spaces in `PATH` explode.

Relevant libtool variables:
~~~sh
# Object dumper program.
OBJDUMP="objdump"

# Method to check whether dependent libraries are shared objects.
deplibs_check_method="file_magic file format (pei*-i386(.*architecture: i386)?|pe-arm-wince|pe-x86-64)"

# Command to use when deplibs_check_method = "file_magic".
file_magic_cmd="\$OBJDUMP -f"

# How to find potential files when deplibs_check_method = "file_magic".
file_magic_glob=""

# Find potential files using nocaseglob when deplibs_check_method = "file_magic".
want_nocaseglob="yes"
~~~

AFAIU it should use `file_magic_cmd="func_win32_libid"`, i.e. command `file` instead of `objdump`. Because is the preferred method in libtool setup, and  `file` is provided via MSYS_ROOT.

dg0yt committed:
Fix dependencies of 'file'

dg0yt committed:
Resolve dependencies at runtime

Don't require msys2 packages to be order from dependent to independent.
This allows to directly use the packages official dependencies, even
if they contain cycles.
It also allows to maintain a stable alphabetical order of the package
declarations, even if the dependencies change over time. This helps to
avoid duplicate declaration.

BillyONeal commented:
I manually cancelled the build because it was stuck trying to build gettext for more than a day

cenit commented on scripts/cmake/vcpkg_configure_make.cmake:
```diff
@@ -203,7 +203,8 @@ function(vcpkg_configure_make)
     # Pre-processing windows configure requirements
     if (VCPKG_TARGET_IS_WINDOWS)
         if(CMAKE_HOST_WIN32)
-            list(APPEND msys_require_packages binutils libtool autoconf automake-wrapper automake1.16 m4 which)
```
which was added recently. Why did you remove it?

dg0yt replied:
It took me a moment to understand which is `which`...
I was taking the msys2 autotools package as a guideline.  And there is `ADDITIONAL_MSYS_PACKAGES`. And from recent discussion I got the impression that `which` is also a shell built-in, but IIUC this is wrong at least for `bash`.

cenit replied:
sorry for the missing code highlighting

dg0yt committed:
Remove duplicate libbz2 entry

dg0yt committed:
Automatically update all msys packages, add autoconf-wrapper

dg0yt committed:
Merge remote-tracking branch 'origin/master' into HEAD

dg0yt committed:
[gettext] Preset more MSVC checks

dg0yt committed:
Update msys2 packages

dg0yt committed:
Fix updating

dg0yt committed:
Merge remote-tracking branch 'origin/master' into HEAD

dg0yt commented:
I will merge master and mark this a ready for review as soon as #30546 is merged.

(Basically it is finished since I resolved the implicit dependencies of `libtool`. But there is no point in re-running CI before #30546 lands.)

dg0yt commented:
@BillyONeal Here is another GH action bot review.

dg0yt committed:
Merge remote-tracking branch 'origin/master' into msys2-refresh

dg0yt committed:
gettext fixes

dg0yt committed:
CI

dg0yt committed:
Merge branch 'microsoft:master' into msys2-refresh

dg0yt committed:
Merge branch 'microsoft:master' into msys2-refresh

dg0yt commented:
@BillyONeal @ras0219-msft I think need this yesterday, not tomorrow.

dg0yt commented:
> @BillyONeal Here is another GH action bot review.

Ping

MonicaLiu0311 commented:
> linux: baseline regression (This PR doesn't affect linux, so the problem shouldn't block the PR.)
> 
> ```
> FAILED: runtime_output_directory/build_id_test 
> : && /usr/bin/c++ -std=c++17 -fno-omit-frame-pointer -ftls-model=initial-exec -fPIC -Wall -Wextra -Wformat-security -Wvla -Wundef -Wmissing-format-attribute -Woverloaded-virtual -Wcast-qual -Wimplicit-fallthrough=5 -Wstringop-truncation -Wsuggest-override -Wmissing-include-dirs -Wextra-semi -Wlogical-op -DSAFE_MUTEX -DENABLED_DEBUG_SYNC -g -Wl,--build-id=sha1 mysys/CMakeFiles/build_id_test.dir/build_id_test.cc.o mysys/CMakeFiles/build_id_test.dir/build_id.cc.o -o runtime_output_directory/build_id_test  -lpthread && cd /mnt/vcpkg-ci/buildtrees/libmysql/x64-linux-dbg/runtime_output_directory && ./build_id_test | egrep -o [0-9a-f]{40} > build_id_test_1 && /mnt/vcpkg-ci/installed/x64-linux/tools/elfutils/bin/eu-readelf -n ./build_id_test | grep "Build ID:" | egrep -o [0-9a-f]{40} > build_id_test_2 && diff build_id_test_1 build_id_test_2
> /mnt/vcpkg-ci/installed/x64-linux/tools/elfutils/bin/eu-readelf: /lib/x86_64-linux-gnu/libdw.so.1: version `ELFUTILS_0.177' not found (required by /mnt/vcpkg-ci/installed/x64-linux/tools/elfutils/bin/eu-readelf)
> ```

Please wait a moment, #30796 will be merged soon, then this PR will be passed.

dg0yt commented:
The functions from this PR are not used on Linux. It is been ready for review since two weeks. I'm waiting for review, not for other PRs.

BillyONeal reviewed (dismissed):
This seems good to me; going to poke others to double check just because all changes to things in scripts are sensitive...

BillyONeal commented:
I will force merge around the Linux baseline issue given that `vcpkg_acquire_msys` as edited here isn't actually used on Linux and the one failure also occurs in most recent full run: https://dev.azure.com/vcpkg/public/_build/results?buildId=87962&view=results

BillyONeal commented on scripts/cmake/vcpkg_acquire_msys.cmake:
```diff
@@ -113,421 +183,456 @@ function(vcpkg_acquire_msys out_msys_root)
-        DEPS mingw-w64-x86_64-expat mingw-w64-x86_64-gcc-libs mingw-w64-x86_64-libiconv
+        URL "https://mirror.msys2.org/mingw/mingw64/mingw-w64-x86_64-openssl-3.1.0-1-any.pkg.tar.zst"
+        SHA512 f146d0f2b31b767422c767f4ea39312cbe90eac86e8b248e24d28595069e6cc450351280dd5b837e27ffdd743fb7eb4880360a658f0d2b3709b8a170be3e37d2
+        #DEPS mingw-w64-x86_64-ca-certificates mingw-w64-x86_64-gcc-libs mingw-w64-x86_64-zlib
```
Any idea why this one is commented out?

dg0yt replied:
I can't say exactly answer why now. I think I wanted to leave (as a comment) what is a declared dependency in vcpkg before this PR, but not a declared mandatory dependency in msys2. Sort-of meant to catch attention during review ;-)

- mingw-w64-x86_64-ca-certificate is an "optional" dependency.
  Cf. https://packages.msys2.org/package/mingw-w64-x86_64-openssl?repo=mingw64
  vcpkg users might opt-in by requesting it as an additional package.
- mingw-w64-x86_64-gcc-libs is not a declared dependency in msys2. I remember that I checked the executables and DLLs of some packages for runtime dependencies, but I can't remember the result for openssl.
- mingw-w64-x86_64-zlib is not a declared dependency in msys2, and in particular also not a declared build time dependency. So this entry is really

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into msys2-refresh

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

BillyONeal commented:
@ras0219-msft and I need to have an argument about whether this goes into a different pipeline or not because this can fail due to no changes in the tree.

dg0yt commented:
> this can fail due to no changes in the tree.

I do not understand exactly. 
`vcpkg_acquire_msys.cmake` should be as reliable as before (apart from new bugs). Actually it should benefit from the simplicitly of more frequent updates.
The test port can and should fail for full CI if vcpkg missed to update a package once in (now) 1.5 years. This is by purpose.
But it wouldn't be triggered in PR CI (unless modifying `vcpkg_acquire_msys.cmake` or ignoring parent hashes).
In that way it acts in CI as needed.

BillyONeal commented:
Robert doesn’t want rerunning the same CI run to fail because the date changed, I think it’s fine for that to happen. The outcome will be either I merge this as is, or split this test port into a separate pipeline so it gets reported differently than ordinary port build failures. You shouldn’t need to further edit anything here.

dg0yt commented:
You may just disable the test port via  `ci-baseline.txt`.
But I need the global scripts and gettext to be unblocked from merge conflicts with other work.

dg0yt commented:
Next incident hitting vcpkg users. Ping @ras0219-msft @BillyONeal.

h3rb-ts commented:
Since this fix has been available for over 30 days, I see the wheels of Microsoft turn very slowly...

dg0yt commented:
> I tried to use this to get past the issue with #31146 but I got the following blocker:

Unrelated. The downloads in `scripts/vcpkg_find_acquire_program(PKGCONFIG).cmake` are up-to-date, in master, but not in `scripts/vcpkg_find_acquire_program.cmake` for "vcpkg-scripts version: c0a38ea06 2022-01-08 (1 year, 4 months ago)".

dg0yt commented:
@h3rb-ts You may need to locally merge this PR with master to get the fix for #31146. If I sync this PR's branch, it will trigger a lengthy CI run with the risk of showing new baseline regressions.

dg0yt committed:
Merge remote-tracking branch 'origin/master' into msys2-refresh

dg0yt committed:
Simplify: Remove update-critical

Desired changes can be picked in git gui etc..

dg0yt committed:
PKGCONFIG moved

dg0yt commented:
Updated to resolve merge conflicts.

dg0yt commented:
If vcpkg in VS is lagging behind the fixes in this repo, vcpkg really needs to catch msys2 updates much earlier, in advance to msys2 package removal.
#31565
#31577

dg0yt commented:
@BillyONeal @ras0219-msft There is only one mirror left (out of eight) providing the outdated/removed grep package, https://mirrors.sjtug.sjtu.edu.cn/msys2/msys/x86_64/grep-3.0-2-x86_64.pkg.tar.xz
It has been two weeks since it was reported in #31446.

> Robert doesn’t want rerunning the same CI run to fail because the date changed,

In the end this is going to happen because the PR was *not* merged, for those who don't have an asset cache.

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

dg0yt commented:
Merging master always means asking for baseline regessions.
podofo adressed in #31658.

BillyONeal commented:
> Merging master always means asking for baseline regessions.

I was going to compare with a recent CI run and merge around that if necessary.

> podofo adressed in #31658.

Thank you!

dg0yt commented:
The `grep is gone` wave of issues is rolling in now.
#31446 
#31808
#31812

I hope it is the last one.
